build-logic/idea/src/main/kotlin/gradlebuild.ide.gradle.kts

```diff
@@ -87,13 +90,36 @@ fun ProjectSettings.configureRunConfigurations() {
     }
 }
 
+fun ProjectSettings.configureTaskTriggers() {
+    taskTriggers {
+        project(":internal-integ-testing").afterEvaluate {
+            afterSync(tasks.named("generateLanguageAnnotations"))
+        }
+    }
+}
```

Comment:
@eskatos can you provide feedback here?

Reply:
I don't recall if the idea-ext plugin is applied to each subproject or not. But reaching out to a subproject here is a smell.
Otherwise this looks like the right way to go to make sure IDEA has the correct state upon import.

Reply:
Maybe the generated annotations could be in a separated source-set that's built by `generateLanguageAnnotations`?

`annotationsSourceSet.srcDir(files(outputDir).builtBy(generateLanguageAnnotations))`

That shouldn't require any special support from IDEs.

Reply:
That's how we do it for precompiled Kotlin script plugins btw, https://github.com/gradle/gradle/blob/f38570a86de2a0fc13d81ecbce23e488081b500b/subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/DefaultPrecompiledScriptPluginsSupport.kt#L344-L356

Reply:
:x: I don't understand why this is needed at all. Everything should be wired in already by `groovy.srcDir(generateLanguageAnnotations.flatMap { it.destDir })` in `internal-integ-testing/build.gradle.kts`.

Can you elaborate why you we need to do more?

In general:
- Never reach into another `project`
- Never use `afterEvaluate`
- Only when there is no other option add some IDE specific configuration (and please add a comment why it is necessary)

We spend a lot of time to clean these things up in the build. Please don't add things like that again.

Reply:
I didn't want to do this either, but @eskatos indicated that it was needed.

https://github.com/gradle/gradle/pull/15645#pullrequestreview-562741738

Reply:
Sorry I did not have that context. The comment is a few days old. Can you check if the problem still exists with the latest changes? In my mind it should work because it should execute the task when it imports the source set. As Rodrigo said it is basically the same as we do in other places.